### PR TITLE
Create @types/desmos

### PR DESCRIPTION
**Pull request by ysulyma**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

Select one of these and delete the others:

If adding a new definition:
- [x] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
- [x] If this is for an npm package, match the name. If not, do not conflict with the name of an npm package.
- [x] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [x] Represents shape of module/library [correctly](https://www.typescriptlang.org/docs/handbook/declaration-files/library-structures.html)
- [ ] `tslint.json` [should contain](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#linter-tslintjson) `{ "extends": "dtslint/dt.json" }`, and no additional rules.
- [x] `tsconfig.json` [should have](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#tsconfigjson) `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

`tslint.json` contains `"npm-naming": false` under `rules`, since the [Desmos npm package](https://www.npmjs.com/package/desmos) (which is not official) is behind the current version of Desmos.

### CODE REVIEW

File: types/desmos/tslint.json [hunk 1]
```diff
@@ -0,0 +1,6 @@
+{
+    "extends": "dtslint/dt.json",
+    "rules": {
+        "npm-naming": false
```
Comment (sandersn) [inline types/desmos/tslint.json:4]: why do you need this? desmos on npm seems to correspond to these types

**Review by sandersn — CHANGES_REQUESTED** [review]

File: types/desmos/tslint.json (hunk 1 above)
Comment (ysulyma) [reply types/desmos/tslint.json:4]: Desmos on npm is at version 1.5.4. These types were produced from the v1.6 documentation, https://www.desmos.com/api/v1.6/docs/index.html.

Comment (ysulyma) [reply types/desmos/tslint.json:4]: @sandersn the `desmos` package on npm (which is not produced or maintained by Desmos) is versioned `1.5.4`. These types were produced for Desmos 1.6, so they should be versioned as `1.6.0`.

Comment (sandersn) [reply types/desmos/tslint.json:4]: Will `@types/desmos` be used in conjunction with the `desmos` package? If so, then the types should cover `1.5.4` or earlier. Definitely Typed is not intended for shipping types ahead of a released version of a package.

If not, then this PR's imports/exports need to change -- right now, they use `import x from 'desmos'`, which only works if `@types/desmos` is matched with `desmos`. If `@types/desmos` is for use from the browser, then everything should be global -- no imports or exports.

Comment (ysulyma) [reply types/desmos/tslint.json:4]: Got it! It is **not** intended to be used in conjunction with the `desmos` package, I've pushed new changes to put everything in the global namespace.

**Review by sandersn — APPROVED** [review]

**Comment by ysulyma** [comment]
Ready to merge

**Pull request merged**